### PR TITLE
Feature/get projects

### PR DESCRIPTION
**Pull request by ncontinanza**

### CODE REVIEW

File: src/api/project/repository/project_repository.go [hunk 1]
```diff
@@ -20,8 +20,32 @@ func NewProjectRepository(db *gorm.DB) domain.Repository {
 	}
 }
 
+func (t *ProjectRepository) GetAll(_ context.Context) ([]*model.Projects, error) {
+	var projects []*model.Projects
+
+	err := t.DB.Order("id desc").Find(&projects).Error
+	if err != nil {
+		log.Printf("Error getting Tasks %v", err)
```
Comment (Mauro-Rizzi) [inline src/api/project/repository/project_repository.go]: Am I understanding this wrong or the log output should be "Error getting Projects %v"?

File: src/api/project/repository/project_repository.go [hunk 2]
```diff
@@ -20,8 +20,32 @@ func NewProjectRepository(db *gorm.DB) domain.Repository {
...
+
+	return projects, nil
+}
+
+func (t *ProjectRepository) GetById(_ context.Context, id string) (*model.Projects, error) {
+	var project *model.Projects
+
+	err := t.DB.First(&project, id).Error
+	if err != nil {
+		log.Printf("Error getting Task %v", err)
```
Comment (Mauro-Rizzi) [inline src/api/project/repository/project_repository.go]: Am I understanding this wrong or the log output should be "Error getting Project %v"?

File: src/api/task/api/dto/task.go [hunk 3]
```diff
@@ -1,51 +1,62 @@
...
 	"time"
 )
 
 // Task dto
 type Task struct {
 	ID             int64     `validate:"gt=0" json:"id"`
-	Name           string    `validate:"required,min=2,max=50" json:"name"`
+	Name           string    `validate:"required,min=2,max=500" json:"name"`
 	Description    string    `validate:"required" json:"description"`
 	StartDate      time.Time `validate:"required" json:"start_date"`
```
Comment (Mauro-Rizzi) [inline src/api/task/api/dto/task.go]: What does start_date stand for here? If it's the creation time of the task why isn't it autogenerated?

File: src/api/task/api/dto/task.go [hunk 4]
```diff
@@ -1,51 +1,62 @@
...
-	HoursWorked    int       `validate:"required" json:"hours_worked"`
+	WorkedHours    int       `validate:"required" json:"worked_hours"`
```
Comment (Mauro-Rizzi) [inline src/api/task/api/dto/task.go]: Why is hours worked a required field? Theoretically we should initialize it at 0 hours upon creation and only modify it when work is logged

File: src/api/task/api/dto/task.go [hunk 5]
```diff
@@ -1,51 +1,62 @@
...
 	EstimatedHours int       `validate:"required" json:"estimated_hours"`
-	ProjectId      int64     `validate:"required" json:"project_id"`
+	ProjectID      int64     `validate:"required" json:"project_id"`
 	State          string    `validate:"required" json:"state"`
 	CreationDate   time.Time `validate:"required" json:"creation_date"`
```
Comment (Mauro-Rizzi) [inline src/api/task/api/dto/task.go]: Why is CreationDate a required field, shouldn't this be optional and simply enforce it's initialization with sysdate when we create a task?

File: src/api/task/api/dto/task.go [hunk 6]
```diff
@@ -1,51 +1,62 @@
...
 	AssignedTo     string    `validate:"required" json:"assigned_to"`
```
Comment (Mauro-Rizzi) [inline src/api/task/api/dto/task.go]: Is it not a possibility to create a task without assigning it to someone? If it's possible we shouldn't make it a required field and we should default to null (Assigned to no one)

File: src/api/task/api/dto/task.go (hunk 4 above)
Comment (ncontinanza) [reply src/api/task/api/dto/task.go]: shrek

File: src/api/task/api/dto/task.go (hunk 5 above)
Comment (ncontinanza) [reply src/api/task/api/dto/task.go]: @AlexDaciuk @julitaras qué opinan?

**Review by Mauro-Rizzi — APPROVED** [review]

File: src/api/task/api/dto/task.go (hunk 5 above)
Comment (AlexDaciuk) [reply src/api/task/api/dto/task.go]: por cuestiones de auditoria tendria que tener un CreationDate, me parece bien que se le planche systime al momento de crearse y listo

File: src/api/task/api/dto/task.go (hunk 6 above)
Comment (AlexDaciuk) [reply src/api/task/api/dto/task.go]: makes sense

File: src/api/task/api/dto/task.go (hunk 4 above)
Comment (AlexDaciuk) [reply src/api/task/api/dto/task.go]: como en la otra consulta, son datos que tienen que estar, me parece bien que al momento de crearse se defaultee a 0

**Pull request merged**